makroer/sykehusopphold.sas

```diff
@@ -0,0 +1,354 @@
+
+
+ 
+### Output
+Makroen lager 12 (eventuelt 14) nye variabler:
+1.  `SHO_nr_pid`: Nummererte SHO-opphold pr pid, dersom SHO best�r av flere opphold har disse samme nummer
+2.  `SHO_id`: Unik ID per sykehusopphold (pid+SHO_nr_pid)
+3.  `SHO_Intern_nr`: Nummerer oppholdene innenfor hver SHO for hver pid
+4.  `SHO_Antall_i_SHO`: Antall opphold i SHO
+5.  `SHO_inndato`: inndato p� f�rste opphold i SHO
+6.  `SHO_utdato`: utdato p� siste opphold i SHO
+7.  [eventuelt] `SHO_inndatotid`: inntid (dato og tidspunkt) p� f�rste opphold i SHO (hvis behold_datotid ulik 0))
+8.  [eventuelt] `SHO_utdatotid`: uttid (dato og tidspunkt) p� siste opphold i SHO (hvis behold_datotid ulik 0))
+9.  `SHO_aar`: �r ved utskriving - alternativt minaar=1 (�r ved f�rste opphold)
+10. `SHO_liggetid`: tidsdifferanse mellom inndatotid p� det f�rste oppholdet og utdatotid p� det siste oppholdet i SHO
+11. `SHO_aktivitetskategori3`: 1 hvis ett av oppholdene er d�gn, eller 2 hvis ett av oppholdene er dag (og ingen d�gn), eller 3 hvis oppholdene er kun poli
+12. `SHO_hastegrad`: 1 hvis ett av oppholdene er akutt, 4 ellers
+13. `SHO_forste_hastegrad`: Hastegrad for f�rste avdelingsopphold. 1 for akutt, 4 for elektiv
+14. `SHO_uttilstand`: max(uttilstand) for oppholdene i en SHO ("som d�d" hvis ett av oppholdene er "som d�d")
+
+
+### Endringslogg
+- 07.09.2021 Opprettet av Janice Shu
+ */
+
+data &dsn;
+  set &dsn;
+  if inntid=. then inntid=0;
+  if uttid=. then uttid=0;
+  inndatotid=dhms(innDato,0,0,innTid);
+  utdatotid=dhms(utDato,0,0,utTid);
+  format inndatotid utdatotid datetime18.;
+run;
+
+proc sort data=&dsn out=&dsn;
+  by pid inndatotid utdatotid;
+run;
+
+/* Lage oppholdsnummer*/
+
+Data &dsn;
+  set &dsn;
+  by pid inndatotid utdatotid;
+  If first.PID=1 then Oppholdsnr=0;
+	Oppholdsnr+1;	
+run;
+
+proc sort data=&dsn;
+  by aktivitetskategori3 pid inndatotid utdatotid;
+run;
+
+/*IDENTIFISERE OVERF�RINGER */
+/* #### improvement for the future.  keep ony necessary variables when making the first tmp dataset */
+
+data Overforinger;
+set &dsn;
+retain flag;
+
+	Lag_PID=Lag(PID);
+	Lag_UtDato=Lag(UtDato);
+	Lag_UtDatotid=Lag(UtDatotid);
+	Lag_BehSh=Lag(BehSh);
+
+  lag_aktkat3=lag(aktivitetskategori3);
+  lag_hastegrad=lag(hastegrad);
+
+  /*overf�ring p� d�gnopphold*/
+  if aktivitetskategori3=1 then do;
+
+	if (PID=LAG_PID) then do;
+
+		/* dager differansen */
+		Dager_mellom=InnDato-LAG_UtDato;
+
+		/* Beregne tidsdifferansen mellom opphold */
+		Sek_mellom=inndatotid-lag_utdatotid; /* Tidsdifferanse (sekunder) mellom inndatotid p� dette oppholdet og utdatotid p� forrige opphold */
+
+		/* increase flag value if there is a longer break between opphold */
+    	if Dager_mellom>1 then do; 
+	 	 flag+1;
+    	end;
+ 
+    end;
+
+	else do; /* increase flag value for different pid */
+	  flag+1;
+	end;
+
+	if BehSH ne LAG_BehSH and sek_mellom>=0 and sek_mellom<=&overforing_tid then Overforing=flag;
```

Comment:
De som har negativ 'sek_mellom' blir ikke inkludert her. Det er endel rader hvor pasienten blir innskrevet på ny avdeling før de er utskrevet på forrige avdeling.
Alternativ:  if BehSH ne LAG_BehSH and sek_mellom ne . and sek_mellom<=&overforing_tid then Overforing=flag;

Reply:
This was the major change from your last test, I thought I had changed it, but apparently not, sorry!  It is now fixed.